### PR TITLE
[Improvement] Added support for authorizing using the Phone

### PR DESCRIPTION
**Pull request by JPZV**
This is by using a generated QR Code on where the calendar should be.

This just add a new method, and `authorize.js` should be working fine when using `Desktop App` as before.

Sadly, the request URL is so big, so it generates a big QR Code, but it should be working just fine even on pretty small (relatively speaking) screens.

### CODE REVIEW

File: node_helper.js [hunk 1]
```diff
@@ -22,6 +23,24 @@ module.exports = NodeHelper.create({
...
+        margin: 5,
+        errorCorrectionLevel: 'H',
+        quality: 0.95,
+      }, (err, qrUrl) => 
+      {
+        if (err) {
+          Log.warn(this.name + ": " + err);
+        }
+        payload.qrUrl = qrUrl;
+        this.sendSocketNotification("AUTH_NEEDED_QR", payload)
```
Comment (randomBrainstormer) [inline node_helper.js:40]: Just to make sure, if there's an error then payload.qrUrl is assigned null/undefined? In case is not a falsey value I'd add a return value after the Log.warn just to be safe : ) 

```suggestion
        if (err) {
          Log.warn(this.name + ": " + err);
          return
        }
        payload.qrUrl = qrUrl;
        this.sendSocketNotification("AUTH_NEEDED_QR", payload)
```

**Review by randomBrainstormer — CHANGES_REQUESTED** [review]
Fantastic job, and thank you for the PR! I really appreciate the effort you put into this. 
The changes look great, and the Pull Request is well-structured. The code changes align perfectly with the PR's purpose. Overall, you've done an excellent job! I only have a few minor comments/suggestions that I'd like you to address before we merge. Once those are resolved, we'll be good to go. Keep up the good work, and thank you for your valuable contribution!

File: README.md [hunk 2]
```diff
@@ -19,14 +19,16 @@ To install the module, use your terminal to:
...
+5. For authorize, you have two options:
+   1. If you've selected `Desktop app` at the step 1, then run this command from the MMM-GoogleCalendar directory: `node authorize.js` and follow the instructions that will display in the console. NOTE: After completing the `authorize.js` script it should print your calendar ID and most recent entries, you can copy the calendar ID to use later in the config file. (If you can't find your ID, check the troubleshooting section for help). If you run the script but don't see anything happening, check the troubleshooting section below, you may need to connect through VNC rather than SSH.
+   2. If you've selected `Web`, then you'll need to scan the QR Code with your phone and open the URL from the same WiFi network. After giving access to your calendar, your phone should open the Magic Mirror interface (Tip, you should temporally give access to your phone to the Magic Mirror by setting `ipWhitelist: []` on `~/MagicMirror/config/config.js`). Generally the calendar ID is your email (e.g. `example@gmail.com`), but if it doesn't work, then you must find it from the Calendar Settings at [Google Calendar](https://calendar.google.com/).
```
[the remark this answers is not in the document]
Comment (JPZV) [reply README.md]: Thanks!!! I'm pretty bad when I try to explain myself in English

File: node_helper.js (hunk 1 above)
Comment (JPZV) [reply node_helper.js]: Thanks for pointing that! But we shouldn't return on that code because we need to send back the `AUTH_NEEDED_QR` notification, otherwise the calendar will be stuck with the "Loading..." screen.

I just committed a new change where, instead of calling return, I just set `payload.qrUrl` to `false`. With this, we should be fine I guess

**Comment by JPZV** [comment]
Thanks!!! I'm glad that you liked the Pull Request.

I resolved both changes, but now I've found a new big issue, and it's that Google doesn't like private IPs neither `.local` top-level domains for Redirect URLs. `http://localhost`, which is what I used while developing this PR, works just fine for Google, but any other method like `192.168.0.20` or `magicmirror.local` will fail with "_must use a domain that is a valid [top private domain](https://github.com/google/guava/wiki/InternetDomainNameExplained#public-suffixes-and-private-domains)_"

The only workaround is to set a false domain on `/etc/host` file from the local machine (not where Magic Mirror lives, but the "client" which scans the QR code), which is relative easy to do in Windows or Linux (and maybe MacOS?), but on many phones that's another thing. For that, you would need to use a DNS Server like Pi.hole, or use an own valid domain (and maybe open/forward ports on the router if the NAT Loopback is not enabled).

So, TL;DR: I don't know if you would want to proceed with this type of authorizing, because it becomes very hackish to do for the end-user (for me and maybe for many developers it's not that hard, I do have a domain and also a Pi.hole server, but I want to think about those random people who came here and are not that technician like us)

**Comment by randomBrainstormer** [comment]
> Thanks!!! I'm glad that you liked the Pull Request.
> 
> I resolved both changes, but now I've found a new big issue, and it's that Google doesn't like private IPs neither `.local` top-level domains for Redirect URLs. `http://localhost`, which is what I used while developing this PR, works just fine for Google, but any other method like `192.168.0.20` or `magicmirror.local` will fail with "_must use a domain that is a valid [top private domain](https://github.com/google/guava/wiki/InternetDomainNameExplained#public-suffixes-and-private-domains)_"
> 
> The only workaround is to set a false domain on `/etc/host` file from the local machine (not where Magic Mirror lives, but the "client" which scans the QR code), which is relative easy to do in Windows or Linux (and maybe MacOS?), but on many phones that's another thing. For that, you would need to use a DNS Server like Pi.hole, or use an own valid domain (and maybe open/forward ports on the router if the NAT Loopback is not enabled).
> 
> So, TL;DR: I don't know if you would want to proceed with this type of authorizing, because it becomes very hackish to do for the end-user (for me and maybe for many developers it's not that hard, I do have a domain and also a Pi.hole server, but I want to think about those random people who came here and are not that technician like us)

Thanks for diving into this. I get the tech hurdles we're facing. Asking users, especially those not super tech-savvy, to mess with the `/etc/host` file or dive into DNS setups can be a real headache. Sure, for pros and network gurus, this might be a walk in the park, but we've got to make things smooth for everyone else too.

Maybe we need to rethink the whole phone authorization thing or whip up a super clear step-by-step guide. Though, gotta admit, it could still be a puzzle for some. Got any ideas ? What if we let users toggle this feature on right from the plugin's settings? That way, we keep the flow smooth and just beef it up for folks wanting a bit more.

**Comment by JPZV** [comment]
Sorry for the delay! I have a crazy week right now!

Yeah! It's a total mess. Thinking about it, this will help to a handful of people actually, specially those who cannot have a keyboard/mouse attached to the mirror (on my case, I fried up the only USB port, and the board is sooooooooooooo slow so I cannot have any type of remote desktop on it), so it could be a good idea to leave it as an optional (disabled by default) setting.

Not sure if doing a guide would help though. I mean, we can link or re-use the documentation from pi.hole for example, but for me, a tech savvy, it took too much effort to make it work, even with the official documentation on hand, and many other guides on internet.

Anyway, I'll try to give it a little rework this weekend, but I'll use this same pull request, as it shouldn't change that much
